llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -3572,9 +3550,38 @@ tryToMatchAndCreateMulAccumulateReduction(VPReductionRecipe *Red,
             ExtCost += Ext1->computeCost(VF, Ctx);
           if (OuterExt)
             ExtCost += OuterExt->computeCost(VF, Ctx);
+          InstructionCost BaseCost = ExtCost + MulCost + RedCost;
 
-          return MulAccCost.isValid() &&
-                 MulAccCost < ExtCost + MulCost + RedCost;
+          if (IsPartialReduction) {
```

Comment:
Same as my comment above, this is now a big change and none of the tests change. My suggestion is to keep the code the same as it was, but to create a new PR to add the asserts that partial reduction cost is valid.